parsers/ES.py

```diff
@@ -163,40 +243,46 @@ def fetch_production(
+
+    ## Production mapping override for Canary Islands
+    # NOTE the LNG terminals are not built yet, so power generated by "gas" or "cc" in ES-CN domain is actually using oil.
+    # Recheck this every 6 months and move to gas key if there has been a change.
+    # Last checked: 2022-06-27
+    if zone_key.split("-")[1] == "CN":
+        fuel_mapping["gas"] = "oil"
+        fuel_mapping["cc"] = "oil"
+
+    if zone_key == "ES-IB-ME" or zone_key == "ES-IB-FO":
+        fuel_mapping["gas"] = "oil"
+    if zone_key == "ES-IB-IZ":
+        fuel_mapping["die"] = "gas"
+
+    island_data = fetch_island_data(zone_key, ses, target_datetime, fuel_mapping)
     productionEventList = ProductionBreakdownList(logger)
-
     for event in island_data:
         production = ProductionMix()
+        for mode in [
+            "biomass",
+            "coal",
+            "gas",
+            "geothermal",
+            "nuclear",
+            "oil",
+            "solar",
+            "wind",
+            "unknown",
+        ]:
+            if mode in event.keys():
+                production.add_value(mode, event[mode])
         storage = StorageMix()
-
-        for mode in PRODUCTION_MAPPING.keys():
-            if mode in event.__dir__():
-                value = getattr(event, mode)
-                ## Production mapping override for Canary Islands
-                # NOTE the LNG terminals are not built yet, so power generated by "gas" or "combined" in ES-CN domain is actually using oil.
-                # Recheck this every 6 months and move to gas key if there has been a change.
-                # Last checked: 2022-06-27
-                fuel_mapping = PRODUCTION_MAPPING
-                if zone_key.split("-")[1] == "CN":
-                    fuel_mapping = {
-                        **PRODUCTION_MAPPING,
-                        "gas": "oil",
-                        "combined": "oil",
-                    }
-                    # Reset gas to 0, as it is now oil, normally this should be skipped or set to None but to conform to how REE does it, we set it to 0
-                    # We should aim to change this upstream in the future to make it uniform with other parsers going forward.
-                    production.gas = 0
-                production.add_value(fuel_mapping[mode], value, True)
-                # Zone specific override for El Hierro
-                # Hydro response is hydro storage, not hydro production
-                if zone_key == "ES-CN-HI" and mode == "hydraulic":
-                    storage.hydro = -value
-                    production.hydro = 0
-
+        if "hydro" in event.keys():
+            storage.hydro = -event["hydro"]
         productionEventList.append(
             zoneKey=zone_key,
-            datetime=datetime.fromtimestamp(event.timestamp).astimezone(timezone.utc),
+            datetime=datetime.fromisoformat(event["timestamp"]).astimezone(
+                timezone.utc
+            ),
```

Comment:
As noted, with the above changes that i suggested datetime.fromisoformat is not needed.